skimage/measure/_structural_similarity.py

```diff
@@ -3,36 +3,137 @@
 __all__ = ['structural_similarity']
 
 import numpy as np
+from scipy.ndimage.filters import uniform_filter, convolve1d
 
 from ..util.dtype import dtype_range
-from ..util.shape import view_as_windows
 
 
-def structural_similarity(X, Y, win_size=7,
-                          gradient=False, dynamic_range=None):
+def gaussian_filter2(X, sigma=1.5, size=11):
+    """ nD Gaussian filter with specific window extent.
+
+    matches the implementation of Wang. et. al.
+
+    Parameters
+    ----------
+    X : ndarray
+        image
+    sigma : float
+        Gaussian standard deviation (pixels)
+    size : float
+        Gaussian kernel extent (pixels)
+
+    Returns
+    -------
+    X : ndarray
+        filtered image
+
+    Notes
+    -----
+    scipy.ndimage.gaussian is very similar, but uses a 13 tap FIR filter
+    rather than the 11 tap one of Wang. et. al.
+    """
+    radius = (size - 1) // 2
+    r = np.arange(2*radius + 1) - radius
+    filt = np.exp(-(r * r)/(2 * sigma * sigma))
+    filt /= filt.sum()
+    for ax in range(X.ndim):
+        X = convolve1d(X, filt, axis=ax)
+    return X
+
+
+def _discard_edges(X, pad):
+    """ Remove border of width pad from ndarray X.
```

Comment:
@JDWarner Can this be done by specifying negative values to np.pad?  Otherwise this should become a generic utility function, since I am pretty sure that `skimage.segmentation.clear_border` needs it too.